### ISSUE TITLE
require_relative issue

### ISSUE THREAD

**Issue by grunxen** [castwide/solargraph#3]
Autocompletion doesn't work for source files that was imported using require_relative.
For example:
```
# a.rb

class A
  def hello_a
  end
end
```

```
# b.rb

require_relative 'a'

class B
  def hello_b
  end
end

b = B.new
b.hello_b # completion provided

a = A.new
a. # completion not provided
```

**Comment by castwide** [comment]
Confirmed. Solargraph::ApiMap needs a mechanism to process require_relative calls.

In the meantime, if you're using the VS Code extension, there's a workaround:
1. Open the folder that contains both files.
2. Create a .yardopts file in the root folder with the following line:
    **/*.rb

This will tell YARD to parse all the scripts in the folder. (The default is to parse the lib and app folders only.) The YARD docs will get updated whenever you save a change. Solargraph uses YARD to complement static analysis, so autocompletion for A and B should become available anywhere in the folder.

**Comment by grunxen** [comment]
> Create a .yardopts file in the root folder with the following line:
**/*.rb
> 
Without this if you try to get suggestions in spec file the server will stuck. Because of mutex another requests will not work too. As a solution I suggest to add this configuration by default. Another solution is to kill thread but it could be dangerous because of side effects.

**Comment by castwide** [comment]
> Without this if you try to get suggestions in spec file the server will stuck. Because of mutex another requests will not work too.

Can you provide details on how to reproduce this bug? If there's a deadlock or an infinite loop occurring somewhere, I'd like to track it down.

I agree that the default set of parsed files should change. The code and API maps should probably include everything in the folder regardless of the YARD configuration. There are some exceptions to consider, but I'm looking into it.

**Comment by castwide** [comment]
Configurable workspace options with sane defaults are committed to the master branch.

Solargraph will parse all .rb files in the current workspace excluding the spec directory. This behavior can be configured with a .solargraph.yml file in the workspace root. You can run `solargraph config` to generate a default configuration file.

I excluded the spec directory by default because it caused serious lag in one of the projects I tested using Solargraph from VS Code. Parsing the tests doesn't seem to provide much benefit anyway, at least when using RSpec. Its DSL makes heavy use of magic methods, and it's not currently possible to get suggestions or documentation for them.

**Comment by castwide** [comment]
The updates to parsing rules and configurable options in .solargraph.yml are published in gem version 0.9.0.
